Docker-compose error

I'm getting an error when trying to run `docker-compose up` on develop and main branch. Says wasm toolchain is not installed. Ive tried a couple things to try and get a quick fix and submit a PR nothing worked so far. Ive tried:

- creating a custom Dockerfile based off https://github.com/paritytech/scripts/blob/master/dockerfiles/contracts-ci-linux/Dockerfile
- using `scssubstratee/substratee_dev:1804-2.12-1.1.3-001` image as seen in `docker-start.sh` script


Complete error message
```
node-template | error: failed to run custom build command for `node-polkadex-runtime v3.0.0 (/var/www/node-template/runtime)`
node-template | 
node-template | Caused by:
node-template |   process didn't exit successfully: `/var/www/node-template/target/release/build/node-polkadex-runtime-f2355782a8411852/build-script-build` (exit status: 1)
node-template |   --- stderr
node-template |   Rust WASM toolchain not installed, please install it!
node-template | 
node-template |   Further error information:
node-template |   ------------------------------------------------------------
node-template |   warning: output filename collision.
node-template |   The bin target `wasm-test` in package `wasm-test v1.0.0 (/tmp/.tmphHPdE1)` has the same output filename as the lib target `wasm_test` in package `wasm-test v1.0.0 (/tmp/.tmphHPdE1)`.
node-template |   Colliding filename is: /tmp/.tmphHPdE1/target/wasm32-unknown-unknown/debug/deps/wasm_test.wasm
node-template |   The output filenames should be unique.
node-template |   Consider changing their names to be unique or compiling them separately.
node-template |   This may become a hard error in the future; see <https://github.com/rust-lang/cargo/issues/6313>.
node-template |   If this looks unexpected, it may be a bug in Cargo. Please file a bug report at
node-template |   https://github.com/rust-lang/cargo/issues/ with as much information as you
node-template |   can provide.
node-template |   cargo 1.54.0-nightly (e51522ab3 2021-05-07) running on `x86_64-unknown-linux-gnu` target `wasm32-unknown-unknown`
node-template |   First unit: Unit { pkg: Package { id: PackageId { name: "wasm-test", version: "1.0.0", source: "/tmp/.tmphHPdE1" }, ..: ".." }, target: TargetInner { name: "wasm-test", doc: true, ..: with_path("/tmp/.tmphHPdE1/src/main.rs", Edition2018) }, profile: Profile { ..: default_dev() }, kind: Target(CompileTarget { name: "wasm32-unknown-unknown" }), mode: Build, features: [], is_std: false, dep_hash: 0 }
node-template |   Second unit: Unit { pkg: Package { id: PackageId { name: "wasm-test", version: "1.0.0", source: "/tmp/.tmphHPdE1" }, ..: ".." }, target: TargetInner { ..: lib_target("wasm_test", ["cdylib"], "/tmp/.tmphHPdE1/src/lib.rs", Edition2018) }, profile: Profile { ..: default_dev() }, kind: Target(CompileTarget { name: "wasm32-unknown-unknown" }), mode: Build, features: [], is_std: false, dep_hash: 0 }
node-template |      Compiling wasm-test v1.0.0 (/tmp/.tmphHPdE1)
node-template |   error[E0463]: can't find crate for `std`
node-template |     |
node-template |     = note: the `wasm32-unknown-unknown` target may not be installed
node-template |     = help: consider downloading the target with `rustup target add wasm32-unknown-unknown`
node-template |     = help: consider building the standard library from source with `cargo build -Zbuild-std`
node-template | 
node-template |   error: aborting due to previous error
node-template | 
node-template |   For more information about this error, try `rustc --explain E0463`.
node-template |   error[E0463]: can't find crate for `std`
node-template |     |
node-template |     = note: the `wasm32-unknown-unknown` target may not be installed
node-template |     = help: consider downloading the target with `rustup target add wasm32-unknown-unknown`
node-template |     = help: consider building the standard library from source with `cargo build -Zbuild-std`
node-template | 
node-template |   error: aborting due to previous error
node-template | 
node-template |   For more information about this error, try `rustc --explain E0463`.
node-template |   error: could not compile `wasm-test`
node-template | 
node-template |   To learn more, run the command again with --verbose.
node-template |   warning: build failed, waiting for other jobs to finish...
node-template |   error: build failed
node-template |   ------------------------------------------------------------
node-template | 
node-template | warning: build failed, waiting for other jobs to finish...
node-template | error: build failed
node-template exited with code 101
```

--- a/2nodes.yml
+++ b/2nodes.yml
@@ -4,7 +4,7 @@ services:
     image: polkadex-node
     container_name: alice
     command: >
-      polkadex-node --base-path /tmp/alice --chain local --alice --port 30333 --ws-port 9945 --rpc-port 9933 --node-key 0000000000000000000000000000000000000000000000000000000000000001 --telemetry-url "wss://telemetry.polkadot.io/submit/ 0" --validator
+      /Polkadex/target/release/polkadex-node --base-path /tmp/alice --chain local --alice --port 30333 --ws-port 9945 --rpc-port 9933 --node-key 0000000000000000000000000000000000000000000000000000000000000001 --telemetry-url "wss://telemetry.polkadot.io/submit/ 0" --validator
     ports:
       - "30333:30333"
       - "9933:9933"
@@ -19,7 +19,7 @@ services:
     image: polkadex-node
     container_name: bob
     command: >
-      polkadex-node --base-path /tmp/bob --chain local --bob --port 30334 --ws-port 9946 --rpc-port 9934 --telemetry-url "wss://telemetry.polkadot.io/submit/ 0" --validator --bootnodes /ip4/172.28.1.1/tcp/30333/p2p/12D3KooWEyoppNCUx8Yx66oV9fJnriXwCcXwDDUA2kj6vnc6iDEp
+      /Polkadex/target/release/polkadex-node --base-path /tmp/bob --chain local --bob --port 30334 --ws-port 9946 --rpc-port 9934 --telemetry-url "wss://telemetry.polkadot.io/submit/ 0" --validator --bootnodes /ip4/172.28.1.1/tcp/30333/p2p/12D3KooWEyoppNCUx8Yx66oV9fJnriXwCcXwDDUA2kj6vnc6iDEp
     ports:
       - "30334:30333"
       - "9934:9933"

--- a/2nodes.yml
+++ b/2nodes.yml
@@ -1,0 +1,61 @@
+version: '1'
+services:
+  node_alice:
+    image: polkadex-node
+    container_name: alice
+    command: >
+      ./Polkadex/target/release/polkadex-node --base-path /tmp/alice \
+        --chain local \
+        --alice \
+        --port 30333 \
+        --ws-port 9945 \
+        --rpc-port 9933 \
+        --node-key 0000000000000000000000000000000000000000000000000000000000000001 \
+        --telemetry-url 'wss://telemetry.polkadot.io/submit/ 0' \
+        --validator
+    ports:
+      - "30333:30333"
+      - "9933:9933"
+      - "9945:9944"
+    volumes:
+      - "polkadot-data-alice:/data"
+    networks:
+      testing_net:
+        ipv4_address: 172.28.1.1
+
+  node_bob:
+    image: polkadex-node
+    container_name: bob
+    command: >
+      ./Polkadex/target/release/polkadex-node \
+        --base-path /tmp/bob \
+        --chain local \
+        --bob \
+        --port 30334 \
+        --ws-port 9946 \
+        --rpc-port 9934 \
+        --telemetry-url 'wss://telemetry.polkadot.io/submit/ 0' \
+        --validator \
+        --bootnodes /ip4/172.28.1.1/tcp/30333/p2p/12D3KooWEyoppNCUx8Yx66oV9fJnriXwCcXwDDUA2kj6vnc6iDEp
+    ports:
+      - "30334:30333"
+      - "9934:9933"
+      - "9946:9944"
+    volumes:
+      - "polkadot-data-bob:/data"
+    links:
+      - "node_alice:alice"
+    networks:
+      testing_net:
+        ipv4_address: 172.28.1.2
+
+volumes:
+  polkadot-data-alice:
+  polkadot-data-bob:
+
+networks:
+  testing_net:
+    ipam:
+      driver: default
+      config:
+        - subnet: 172.28.0.0/16

--- a/2nodes.yml
+++ b/2nodes.yml
@@ -1,4 +1,4 @@
-version: '1'
+version: '3'
 services:
   node_alice:
     image: polkadex-node

--- a/2nodes.yml
+++ b/2nodes.yml
@@ -4,15 +4,7 @@ services:
     image: polkadex-node
     container_name: alice
     command: >
-      ./Polkadex/target/release/polkadex-node --base-path /tmp/alice \
-        --chain local \
-        --alice \
-        --port 30333 \
-        --ws-port 9945 \
-        --rpc-port 9933 \
-        --node-key 0000000000000000000000000000000000000000000000000000000000000001 \
-        --telemetry-url 'wss://telemetry.polkadot.io/submit/ 0' \
-        --validator
+      polkadex-node --base-path /tmp/alice --chain local --alice --port 30333 --ws-port 9945 --rpc-port 9933 --node-key 0000000000000000000000000000000000000000000000000000000000000001 --telemetry-url "wss://telemetry.polkadot.io/submit/ 0" --validator
     ports:
       - "30333:30333"
       - "9933:9933"
@@ -27,16 +19,7 @@ services:
     image: polkadex-node
     container_name: bob
     command: >
-      ./Polkadex/target/release/polkadex-node \
-        --base-path /tmp/bob \
-        --chain local \
-        --bob \
-        --port 30334 \
-        --ws-port 9946 \
-        --rpc-port 9934 \
-        --telemetry-url 'wss://telemetry.polkadot.io/submit/ 0' \
-        --validator \
-        --bootnodes /ip4/172.28.1.1/tcp/30333/p2p/12D3KooWEyoppNCUx8Yx66oV9fJnriXwCcXwDDUA2kj6vnc6iDEp
+      polkadex-node --base-path /tmp/bob --chain local --bob --port 30334 --ws-port 9946 --rpc-port 9934 --telemetry-url "wss://telemetry.polkadot.io/submit/ 0" --validator --bootnodes /ip4/172.28.1.1/tcp/30333/p2p/12D3KooWEyoppNCUx8Yx66oV9fJnriXwCcXwDDUA2kj6vnc6iDEp
     ports:
       - "30334:30333"
       - "9934:9933"

--- a/README.md
+++ b/README.md
@@ -67,6 +67,100 @@ When running a node via the binary file, data is stored in a local directory typ
 
 This will remove the data folder, note that all chain data is now lost.
 
+## Run as a validator node
+
+```bash
+# Purge any chain data from previous runs
+./target/release/polkadex-node purge-chain --base-path /tmp/alice --chain local
+# Start Alice's node
+./target/release/polkadex-node --base-path /tmp/alice \
+  --chain local \
+  --alice \
+  --port 30333 \
+  --ws-port 9945 \
+  --rpc-port 9933 \
+  --node-key 0000000000000000000000000000000000000000000000000000000000000001 \
+  --telemetry-url 'wss://telemetry.polkadot.io/submit/ 0' \
+  --validator
+```
+
+```bash
+2021-06-30 08:12:38 Polkadex Node    
+2021-06-30 08:12:38 ✌️  version 3.0.0-6426a73b-x86_64-macos    
+2021-06-30 08:12:38 ❤️  by Substrate DevHub <https://github.com/substrate-developer-hub>, 2017-2021    
+2021-06-30 08:12:38 📋 Chain specification: Local Testnet    
+2021-06-30 08:12:38 🏷 Node name: Alice    
+2021-06-30 08:12:38 👤 Role: AUTHORITY    
+2021-06-30 08:12:38 💾 Database: RocksDb at /tmp/alice/chains/local_testnet/db    
+2021-06-30 08:12:38 ⛓  Native runtime: node-polkadex-265 (node-polkadex-1.tx2.au10)    
+2021-06-30 08:12:39 🔨 Initializing Genesis block/state (state: 0xbe0a…5ef3, header-hash: 0xa55f…7888)    
+2021-06-30 08:12:39 👴 Loading GRANDPA authority set from genesis on what appears to be first startup.    
+2021-06-30 08:12:39 ⏱  Loaded block-time = 3s from genesis on first-launch    
+2021-06-30 08:12:39 👶 Creating empty BABE epoch changes on what appears to be first startup.    
+2021-06-30 08:12:39 Using default protocol ID "sup" because none is configured in the chain specs    
+2021-06-30 08:12:39 🏷 Local node identity is: 12D3KooWEyoppNCUx8Yx66oV9fJnriXwCcXwDDUA2kj6vnc6iDEp    
+2021-06-30 08:12:39 📦 Highest known block at #0    
+2021-06-30 08:12:39 〽️ Prometheus server started at 127.0.0.1:9615    
+2021-06-30 08:12:39 Listening for new connections on 127.0.0.1:9945.    
+2021-06-30 08:12:39 👶 Starting BABE Authorship worker    
+2021-06-30 08:12:44 💤 Idle (0 peers), best: #0 (0xa55f…7888), finalized #0 (0xa55f…7888), ⬇ 0 ⬆ 0    
+2021-06-30 08:12:49 💤 Idle (0 peers), best: #0 (0xa55f…7888), finalized #0 (0xa55f…7888), ⬇ 0 ⬆ 0    
+2021-06-30 08:12:54 💤 Idle (0 peers), best: #0 (0xa55f…7888), finalized #0 (0xa55f…7888), ⬇ 0 ⬆ 0
+```
+
+Local node identity is: 12D3KooWEyoppNCUx8Yx66oV9fJnriXwCcXwDDUA2kj6vnc6iDEp shows the Peer ID that Bob will need when booting from Alice's node. This value was determined by the --node-key that was used to start Alice's node.
+
+Now that Alice's node is up and running, Bob can join the network by bootstrapping from her node.
+```bash
+./target/release/polkadex-node purge-chain --base-path /tmp/bob --chain local
+./target/release/polkadex-node \
+  --base-path /tmp/bob \
+  --chain local \
+  --bob \
+  --port 30334 \
+  --ws-port 9946 \
+  --rpc-port 9934 \
+  --telemetry-url 'wss://telemetry.polkadot.io/submit/ 0' \
+  --validator \
+  --bootnodes /ip4/127.0.0.1/tcp/30333/p2p/12D3KooWEyoppNCUx8Yx66oV9fJnriXwCcXwDDUA2kj6vnc6iDEp
+```
+
+If all is going well, after a few seconds, the nodes should peer together and start producing blocks. You should see some lines like the following in the console that started Alice node.
+
+```bash
+2021-06-30 08:16:52 Polkadex Node    
+2021-06-30 08:16:52 ✌️  version 3.0.0-6426a73b-x86_64-macos    
+2021-06-30 08:16:52 ❤️  by Substrate DevHub <https://github.com/substrate-developer-hub>, 2017-2021    
+2021-06-30 08:16:52 📋 Chain specification: Local Testnet    
+2021-06-30 08:16:52 🏷 Node name: Bob    
+2021-06-30 08:16:52 👤 Role: AUTHORITY    
+2021-06-30 08:16:52 💾 Database: RocksDb at /tmp/bob/chains/local_testnet/db    
+2021-06-30 08:16:52 ⛓  Native runtime: node-polkadex-265 (node-polkadex-1.tx2.au10)    
+2021-06-30 08:16:52 🔨 Initializing Genesis block/state (state: 0xbe0a…5ef3, header-hash: 0xa55f…7888)    
+2021-06-30 08:16:52 👴 Loading GRANDPA authority set from genesis on what appears to be first startup.    
+2021-06-30 08:16:52 ⏱  Loaded block-time = 3s from genesis on first-launch    
+2021-06-30 08:16:52 👶 Creating empty BABE epoch changes on what appears to be first startup.    
+2021-06-30 08:16:52 Using default protocol ID "sup" because none is configured in the chain specs    
+2021-06-30 08:16:52 🏷 Local node identity is: 12D3KooWRHDuuHg5ZQcJhvVDKud9XkFz2Dcs2GQKF9KKuTD6quq7    
+2021-06-30 08:16:53 📦 Highest known block at #0    
+2021-06-30 08:16:53 Listening for new connections on 127.0.0.1:9946.    
+2021-06-30 08:16:53 👶 Starting BABE Authorship worker    
+2021-06-30 08:16:53 🔍 Discovered new external address for our node: /ip4/127.0.0.1/tcp/30334/p2p/12D3KooWRHDuuHg5ZQcJhvVDKud9XkFz2Dcs2GQKF9KKuTD6quq7    
+2021-06-30 08:16:53 🔍 Discovered new external address for our node: /ip4/192.168.1.37/tcp/30334/p2p/12D3KooWRHDuuHg5ZQcJhvVDKud9XkFz2Dcs2GQKF9KKuTD6quq7    
+2021-06-30 08:16:53 Creating inherent data providers took more time than we had left for the slot.    
+2021-06-30 08:16:54 🙌 Starting consensus session on top of parent 0xa55fa19cc37ca1f8d93bc06ca1f6fee767f18200516d9e349938601a3fe97888    
+2021-06-30 08:16:54 🎁 Prepared block for proposing at 1 [hash: 0x2959db5e42a7192434d3699d335e5d920da73409963e3081ad43afd93a8cdb4b; parent_hash: 0xa55f…7888; extrinsics (1): [0x4431…4eff]]    
+2021-06-30 08:16:54 🔖 Pre-sealed block for proposal at 1. Hash now 0x5263ed1cbf1b4edbc887cc87786471819cd0614d8aeaff3a898c0c3ffda245c2, previously 0x2959db5e42a7192434d3699d335e5d920da73409963e3081ad43afd93a8cdb4b.    
+2021-06-30 08:16:54 👶 New epoch 0 launching at block 0x5263…45c2 (block slot 541685138 >= start slot 541685138).    
+2021-06-30 08:16:54 👶 Next epoch starts at slot 541685338    
+2021-06-30 08:16:54 ✨ Imported #1 (0x5263…45c2)    
+2021-06-30 08:16:57 🙌 Starting consensus session on top of parent 0x5263ed1cbf1b4edbc887cc87786471819cd0614d8aeaff3a898c0c3ffda245c2    
+2021-06-30 08:16:57 🎁 Prepared block for proposing at 2 [hash: 0x0c513e39a88bcb03d113a18ed824bcbaab03881e9dcdeedbe12e71955dcfe05d; parent_hash: 0x5263…45c2; extrinsics (1): [0xf06a…bf04]]    
+2021-06-30 08:16:57 🔖 Pre-sealed block for proposal at 2. Hash now 0x4293ecd46db852f5add54a24acfcd1ea12f6c26d5470b61736d7cf0e039e3e39, previously 0x0c513e39a88bcb03d113a18ed824bcbaab03881e9dcdeedbe12e71955dcfe05d.    
+2021-06-30 08:16:57 ✨ Imported #2 (0x4293…3e39)    
+2021-06-30 08:16:58 💤 Idle (1 peers), best: #2 (0x4293…3e39), finalized #0 (0xa55f…7888), ⬇ 1.7kiB/s ⬆ 1.9kiB/s 
+```
+
 ## Connecting Polkadot JS Apps to a Local Polkadex Node
 The development node is a Substrate-based node, so you can interact with it using standard Substrate tools. The two provided RPC endpoints are:
 - HTTP: `http://127.0.0.1:9933`

--- a/README.md
+++ b/README.md
@@ -175,6 +175,15 @@ If all is going well, after a few seconds, the nodes should peer together and st
 2021-06-30 08:16:58 💤 Idle (1 peers), best: #2 (0x4293…3e39), finalized #0 (0xa55f…7888), ⬇ 1.7kiB/s ⬆ 1.9kiB/s 
 ```
 
+## Using docker
+
+The following commands will setup a local polkadex network made of 2 nodes. It's using the node key (0000000000000000000000000000000000000000000000000000000000000001). But you should generate your own node key using the subkey as the above.
+
+```bash
+docker build . -t polkadex-node
+docker-compose -f 2nodes.yml up --force-recreate
+```
+
 ## Connecting Polkadot JS Apps to a Local Polkadex Node
 The development node is a Substrate-based node, so you can interact with it using standard Substrate tools. The two provided RPC endpoints are:
 - HTTP: `http://127.0.0.1:9933`

--- a/README.md
+++ b/README.md
@@ -30,11 +30,52 @@ The value we provide to the Substrate community is,
 3. They get a decentralized trading platform to trade Polkadot & Ethereum tokens.
 4. This will be one of the first Decentralized exchanges to have High-Frequency Trading bot support using APIs directly from full nodes.
    ![Web3 Grants](https://github.com/Polkadex-Substrate/Documentation/blob/master/images/web3%20foundation_grants_badge_black.svg)
-## Run the node :dancer:
-You need to have docker installed to run Polkadex. Use the command given below.
+## Build the Polkadex Node 💃
+
+To build Polkadex, you will need a proper Substrate development environment. If you need a refresher setting up your Substrate environment, see [Substrate's Getting Started Guide](https://substrate.dev/docs/en/knowledgebase/getting-started/).
+
+Note that cloning master might result in an unstable build. If you want a stable version, check out the [latest releases](https://github.com/Polkadex-Substrate/Polkadex/releases).
+
+```bash
+# Fetch the code
+git clone https://github.com/Polkadex-Substrate/Polkadex.git
+cd Polkadex
+
+# Build the node (The first build will be long (~30min))
+cargo build --release
 ```
-docker-compose up
+
+If a cargo not found error shows up in the terminal, manually add Rust to your system path (or restart your system):
+
+```bash
+source $HOME/.cargo/env
 ```
+
+Then, you will want to run the node in dev mode using the following command:
+
+```bash
+./target/release/polkadex-node --dev
+```
+
+> For people not familiar with Substrate, the --dev flag is a way to run a Substrate-based node in a single node developer configuration for testing purposes. You can learn more about `--dev` in [this Substrate tutorial](https://substrate.dev/docs/en/tutorials/create-your-first-substrate-chain/interact).
+
+When running a node via the binary file, data is stored in a local directory typically located in ~/.local/shared/polkadex-node/chains/development/db. If you want to start a fresh instance of the node, you can either delete the content of the folder, or run the following command inside the polkadex folder:
+
+```bash
+./target/release/node-polkadex purge-chain --dev
+```
+
+This will remove the data folder, note that all chain data is now lost.
+
+## Connecting Polkadot JS Apps to a Local Polkadex Node
+The development node is a Substrate-based node, so you can interact with it using standard Substrate tools. The two provided RPC endpoints are:
+- HTTP: `http://127.0.0.1:9933`
+- WS: `ws://127.0.0.1:9944`
+
+Start by connecting to it with Polkadot JS Apps. Open a browser to: https://polkadot.js.org/apps/#/explorer. This will open Polkadot JS Apps, which automatically connects to Polkadot MainNet.
+
+Click on the top left corner to open the menu to configure the networks, and then navigate down to open the Development sub-menu. In there, you will want to toggle the "Local Node" option, which points Polkadot JS Apps to ws://127.0.0.1:9944. Next, select the Switch button, and the site should connect to your Polkadex development node.
+
 ## Documentation :books:
 For Tutorials, Documentation and API Reference please check this [page](https://github.com/Polkadex-Substrate/Documentation)
 ## Contribute :heart_eyes:

--- a/README.md
+++ b/README.md
@@ -77,7 +77,6 @@ brew install jq
 
 - Generate node key using `subkey`
 ```bash
-subkey generate --scheme Ed25519 | jq -r '.secretSeed'
 Alice_Node_Key=$(subkey generate --scheme Ed25519 --output-type Json | jq -r '.secretSeed')
 ```
 

--- a/README.md
+++ b/README.md
@@ -69,9 +69,24 @@ This will remove the data folder, note that all chain data is now lost.
 
 ## Run as a validator node
 
+- Install `subkey`, `jq`
+```bash
+curl https://getsubstrate.io -sSf | bash -s --
+brew install jq
+```
+
+- Generate node key using `subkey`
+```bash
+subkey generate --scheme Ed25519 | jq -r '.secretSeed'
+Alice_Node_Key=$(subkey generate --scheme Ed25519 --output-type Json | jq -r '.secretSeed')
+```
+
+- Run Alice's node
+
 ```bash
 # Purge any chain data from previous runs
 ./target/release/polkadex-node purge-chain --base-path /tmp/alice --chain local
+
 # Start Alice's node
 ./target/release/polkadex-node --base-path /tmp/alice \
   --chain local \
@@ -79,7 +94,7 @@ This will remove the data folder, note that all chain data is now lost.
   --port 30333 \
   --ws-port 9945 \
   --rpc-port 9933 \
-  --node-key 0000000000000000000000000000000000000000000000000000000000000001 \
+  --node-key $Alice_Node_Key \
   --telemetry-url 'wss://telemetry.polkadot.io/submit/ 0' \
   --validator
 ```

--- a/dockerfile
+++ b/dockerfile
@@ -18,14 +18,14 @@ RUN cd Polkadex && \
 
 CMD cp ./Polkadex/target/release/polkadex-node /usr/local/bin
 
-RUN useradd -m -u 1000 -U -s /bin/sh -d /polkadex polkadex && \
-	mkdir -p /polkadex/.local/share && \
+RUN useradd -m -u 1000 -U -s /bin/sh -d /polkadex-node polkadex-node && \
+	mkdir -p /polkadex-node/.local/share && \
 	mkdir /data && \
-	chown -R polkadex:polkadex /data && \
-	ln -s /data /polkadex/.local/share/polkadex && \
+	chown -R polkadex-node:polkadex-node /data && \
+	ln -s /data /polkadex-node/.local/share/polkadex-node && \
 	rm -rf /usr/bin /usr/sbin
 
-USER polkadex
+USER polkadex-node
 EXPOSE 30333 9933 9944
 VOLUME ["/data"]
 

--- a/dockerfile
+++ b/dockerfile
@@ -1,0 +1,20 @@
+FROM ubuntu
+RUN apt-get update
+RUN apt-get install -y git && apt-get install -y curl
+RUN git clone https://github.com/Polkadex-Substrate/Polkadex
+RUN cd Polkadex && \
+    git checkout develop && \
+    ls && \
+    apt-get install -y build-essential && \
+    apt-get install -y clang && \
+    apt-get install -y jq && \
+    curl https://sh.rustup.rs -sSf | sh -s -- -y && \
+	export PATH="$PATH:$HOME/.cargo/bin" && \
+	rustup toolchain install nightly-2021-05-11 && \
+	rustup target add wasm32-unknown-unknown --toolchain nightly-2021-05-11 && \
+	rustup default stable && \
+	cargo update -p sp-std --precise e5437efefa82bd8eb567f1245f0a7443ac4e4fe7 && \
+	cargo +nightly-2021-05-11 build --release "--$PROFILE"
+
+
+EXPOSE 30333 9933 9944

--- a/dockerfile
+++ b/dockerfile
@@ -16,17 +16,8 @@ RUN cd Polkadex && \
 	cargo update -p sp-std --precise e5437efefa82bd8eb567f1245f0a7443ac4e4fe7 && \
 	cargo +nightly-2021-05-11 build --release "--$PROFILE"
 
-CMD cp ./Polkadex/target/release/polkadex-node /usr/local/bin
+CMD cp /Polkadex/target/release/polkadex-node /usr/local/bin
 
-RUN useradd -m -u 1000 -U -s /bin/sh -d /polkadex-node polkadex-node && \
-	mkdir -p /polkadex-node/.local/share && \
-	mkdir /data && \
-	chown -R polkadex-node:polkadex-node /data && \
-	ln -s /data /polkadex-node/.local/share/polkadex-node && \
-	rm -rf /usr/bin /usr/sbin
-
-USER polkadex-node
 EXPOSE 30333 9933 9944
-VOLUME ["/data"]
 
-CMD ["/usr/local/bin/polkadex-node"]
+CMD ["/Polkadex/target/release/polkadex-node"]

--- a/dockerfile
+++ b/dockerfile
@@ -16,8 +16,6 @@ RUN cd Polkadex && \
 	cargo update -p sp-std --precise e5437efefa82bd8eb567f1245f0a7443ac4e4fe7 && \
 	cargo +nightly-2021-05-11 build --release "--$PROFILE"
 
-CMD cp /Polkadex/target/release/polkadex-node /usr/local/bin
-
 EXPOSE 30333 9933 9944
 
 CMD ["/Polkadex/target/release/polkadex-node"]

--- a/dockerfile
+++ b/dockerfile
@@ -16,5 +16,17 @@ RUN cd Polkadex && \
 	cargo update -p sp-std --precise e5437efefa82bd8eb567f1245f0a7443ac4e4fe7 && \
 	cargo +nightly-2021-05-11 build --release "--$PROFILE"
 
+CMD cp ./Polkadex/target/release/polkadex-node /usr/local/bin
 
+RUN useradd -m -u 1000 -U -s /bin/sh -d /polkadex polkadex && \
+	mkdir -p /polkadex/.local/share && \
+	mkdir /data && \
+	chown -R polkadex:polkadex /data && \
+	ln -s /data /polkadex/.local/share/polkadex && \
+	rm -rf /usr/bin /usr/sbin
+
+USER polkadex
 EXPOSE 30333 9933 9944
+VOLUME ["/data"]
+
+CMD ["/usr/local/bin/polkadex-node"]

--- a/pallets/pdex-migration/README.md
+++ b/pallets/pdex-migration/README.md
@@ -1,0 +1,144 @@
+## PDEX Migration pallet
+
+### Importing a `pdex-migration` pallet
+
+The first thing you need to do to add the `pdex-migration` pallet is to import the `pdex-migration` crate in your runtime's Cargo.toml file
+
+`runtime/Cargo.toml`
+
+```
+[dependencies]
+#--snip--
+pallet-verifier-lightclient = { git = "https://github.com/Polkadex-Substrate/polkadot-ethereum", default-features = false, rev = "6ac2106a608bf11c53981b67a106c2afd43bbee6" }
+pallet-eth-dispatch = { package = "artemis-dispatch", git = "https://github.com/Polkadex-Substrate/polkadot-ethereum", default-features = false, rev = "6ac2106a608bf11c53981b67a106c2afd43bbee6" }
+pallet-basic-channel = { package = "artemis-basic-channel", git = "https://github.com/Polkadex-Substrate/polkadot-ethereum", default-features = false, rev = "6ac2106a608bf11c53981b67a106c2afd43bbee6" }
+artemis-core = { git = "https://github.com/Polkadex-Substrate/polkadot-ethereum", default-features = false, rev = "6ac2106a608bf11c53981b67a106c2afd43bbee6" }
+
+erc20-pdex-migration-pallet = { path = "https://github.com/Polkadex-Substrate/Polkadex/pallets/pdex-migration", default-features = false }
+```
+
+Add the following two lines to the runtime's std feature.
+
+```
+[features]
+default = ['std']
+std = [
+    #--snip--
+    'pallet-verifier-lightclient/std',
+    'pallet-eth-dispatch/std',
+    'pallet-basic-channel/std',
+    'artemis-core/std',
+    'erc20-pdex-migration-pallet/std',
+    #--snip--
+]
+```
+
+### Adding the `pdex-migration` pallet
+
+`runtime/src/lib.rs`
+
+- Adding to the construct_runtime! Macro
+
+```
+    Currencies: orml_currencies::{Pallet, Call, Event<T>},
+    Dispatch: pallet_eth_dispatch::{Pallet, Call, Storage, Event<T>, Origin},
+    ERC20PDEX: erc20_pdex_migration_pallet::{Pallet, Call, Storage, Config, Event<T>}
+```
+
+- Runtime Configuration
+
+```
+pub use artemis_core::MessageId;
+use pallet_eth_dispatch::EnsureEthereumAccount;
+pub use polkadex_primitives::{Balance};
+
+pub struct CallFilter;
+
+impl Filter<Call> for CallFilter {
+    fn filter(call: &Call) -> bool {
+        match call {
+            Call::ERC20PDEX(_) => true,
+            _ => false
+        }
+    }
+}
+
+impl pallet_eth_dispatch::Config for Runtime {
+    type Origin = Origin;
+    type Event = Event;
+    type MessageId = MessageId;
+    type Call = Call;
+    type CallFilter = CallFilter;
+}
+
+impl erc20_pdex_migration_pallet::Config for Runtime{
+    type Event = Event;
+    type Balance = Balance;
+    type Currency = Currencies;
+    type CallOrigin = EnsureEthereumAccount;
+}
+```
+
+### Genesis Configuration
+
+- Generate EthereumHeader using `eth-relayer` (https://github.com/Polkadex-Substrate/eth-relayer)
+
+```sh
+./build/polkadex-eth-relay getblock
+INFO[0002] Connected to chain                            chainID=3 endpoint="wss://ropsten.infura.io/ws/v3/"
+
+EthereumHeader {
+        parent_hash: hex!("30fe8b0a52f72f1e38556a0a32c3799e285855bcee53048935981028fcc49d44").into(),
+        ...
+}
+```
+
+- Update `initial_difficulty` for the certain network
+
+`node/src/chain_spec.rs`
+Inside the testnet_genesis function we need to add our pallet's configuration to the returned GenesisConfig object as followed:
+```
+    GenesisConfig {
+        /* --snip-- */
+
+        /*** Add This Block ***/
+        // This is Ropsten Config
+        pallet_verifier_lightclient: VerifierLightclientConfig {
+            initial_header: EthereumHeader {
+                parent_hash: hex!("c75694f43b710d53e3026151ecd910b4d1614ff6be90bea0e9e25c71d31ddc94").into(),
+                timestamp: 1624172254u64.into(),
+                number: 10473724u64.into(),
+                author: hex!("1cffe205e97976bb9d1ec006f5222360a89353e0").into(),
+                transactions_root: hex!("9e298e62573bb9fb4d774f48aacfef0299b5b2c711708e4c0966eaa3a297d507").into(),
+                ommers_hash: hex!("1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347").into(),
+                extra_data: hex!("d683010a02846765746886676f312e3136856c696e7578").into(),
+                state_root: hex!("a7c3b4cda608f234e081945534f2c921ea82ee50ec1e5ec3117edf80e7e1d24b").into(),
+                receipts_root: hex!("a6b13691149babc76431bc992102b658e2992ce48923acd830f189ee95838d2a").into(),
+                logs_bloom: (&hex!("400202000000000000100000800000000000000000200000000000004000000a000400000100011000000040000000002000000000000800000000000420000000080000020100000000000800000002000004000200000000000000000020000400000802040000000000880000280000008000000000000000003000000000200000020002000020000a000100000040008400000080000000000200000200320000100000000020004004220000000000000000000000020000000000000100000002000000000000a00000000000000001000100000000000400000020000012000020100000000000000000002000000080000000000000001000010010")).into(),
+                gas_used: 975581u64.into(),
+                gas_limit: 8000000u64.into(),
+                difficulty: 526959644u64.into(),
+                seal: vec![
+                    hex!("a0ef29b20dc8f835f811fd431be5af023ca83b9ab403838404ad09a86e4e27a52f").to_vec(),
+                    hex!("88124e2d4f2bc1eda9").to_vec(),
+                ],
+            },
+            initial_difficulty: 19755084633726428633088u128.into(),
+        },
+        basic_channel_inbound: BasicInboundChannelConfig {
+            source_channel: hex!["EE9170ABFbf9421Ad6DD07F6BDec9D89F2B581E0"].into(),
+        },
+        erc20_pdex_migration_pallet: ERC20PDEXConfig {
+            address: hex!["3f0839385DB9cBEa8E73AdA6fa0CFe07E321F61d"].into()
+        },
+        /*** End Added Block ***/
+    }
+```
+
+### Dispatchable functions
+
+- `mint()`
+
+First, it checks if the function is called by `pallet_eth_dispatch` pallet. If not, it throws the `DispatchError` error.
+Then, it mints `amount` of `AssetId:POLKADEX` tokens to the given `recipient` Polkadex address.
+Finally, it emits `NativePDEXMinted` event with the various parameters.

